index.md

```diff
@@ -8,6 +8,469 @@ This is some welcome text.
+Students are expected to not return anything in the __init__ method.
+
+```python
+class Animal:
+    """An animal in the zoo.
+    """
+    
+    def __init__(self, name):
+        self.name = name
+        return True
+        
+    def is_animal(self):
+        """
+        @rtype: bool
+        """
+        return isinstance(self.name, str)
+        
+    def getter(self):
+        """
+        @rtype: str
+        """
+        return self.name
+```
+
+### E0102 Function redefined {#E0102}
+
+This error occurs when a function, class or method is redefined. If you are 
+getting this error, you should make sure all the functions, methods and classes 
+have different names.
+
+```python
+def pos(obj):
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    if obj < 0:
+        return False
+    else:
+        return True
+        
+        
+def pos(obj):
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    if obj < 1:
+        return False
+    else:
+        return True
+```
+
+### E0103 Not in loop {#E0103}
+
+This error occurs when break and continue keywords are used outside loops. If 
+you are getting this error, please check your code carefully.
+
+```python
+def add(lst):
+    """ Calculates the sum of the elements in the given list.
+    
+    @type lst: list
+    @rtype: int
+    """
+    temp = 0
+    for item in lst:
+        temp += item
+    break # Error on this line
+    return temp
+```
+
+### E0104 Return outside function {#E0104}
+
+This error occurs when a "return" statement is found outside a function or 
+method. If you are getting this error, please check the indentation levels in 
+your code carefully.
+
+```python
+def add(lst):
+    """ Calculates the sum of the elements in the given list.
+    
+    @type lst: list
+    @rtype: bool
+    """
+    temp = 0
+    for item in lst:
+        temp += item
+        
+return False # Error on this line
+```
+
+### E0107 Nonexistent operator {#E0107}
+
+### E0108 Duplicate argument name {E0108}
+
+This error occurs if there are duplicate argument names in function 
+definitions. If you are getting this error, please make sure all the arguments 
+have distinct names.
+
+```python
+def add(lst, lst):
+    """
+    @type lst: list
+    @type lst: list
+    @rtype: int
+    """
+    temp = 0
+    for item in lst:
+        temp += item
+    return temp
+```
+
+### E0110 Abstract class instantiated {#E0110}
+
+### W0101 Unreachable {#W0101}
+
+This error occurs when there is some code behind a "return" or "raise" 
+statement. This code will never be accessed, therefore check your code 
+carefully if you are getting this error. 
+
+```python
+def add(lst):
+    """
+    @type lst: list
+    @rtype: int
+    """
+    temp = 0
+    for item in lst:
+        temp += item
+    return temp
+    temp += 1 # Error on this line
+```
+
+### W0102 Dangerous default value {#W0102}
+
+This error occurs when a mutable value such as a list or dictionary is given a 
```

Comment:
It is *extremely* valuable here to explain why this is an error here, because it is a little subtle.